### ISSUE TITLE
Add new lifecycle event `urlUpdated`

### ISSUE THREAD

**Issue by hirasso** [swup/swup#663]
```js
if (!isHistoryVisit) {
	createHistoryRecord(url + (this.scrollToElement || ''));
}
// Add this event
this.triggerEvent('urlUpdated', { from: this.currentPageUrl, to: url });

this.currentPageUrl = getCurrentUrl();
```

**Comment by daun** [comment]
Definitely makes sense to have this to bridge the gap between link visits and popstate visits.

Adding context for the event is a great idea, I think we should explore this for all events. I'm currently working on a branch that will pass a context object which has the events in a `originalEvent` property. That way, we'd have the best of both worlds. Probably a major version as it'd be a breaking change.


```js
swup.on('clickLink', ({ originalEvent, linkEl }) => {
    console.log(originalEvent.clientX, originalEvent.clientY)
})
```

**Comment by daun** [comment]
Implemented in PR #685
